packages/itwin/property-grid/src/components/PropertyGrid.tsx

```diff
@@ -303,6 +294,24 @@ export const PropertyGrid = ({
           }
         }
 
+        // Do any overrides on default menu options
+        if(defaultContextMenuOptions?.size && defaultContextMenuOptions.size > 0 ) {
+          for(const key of Object.values(PropertyGridDefaultContextMenuKey)) {          
+            const overrides = defaultContextMenuOptions?.get(key);
+            const itemIndex = items.map((item) => item.key).indexOf(key);
+            items[itemIndex] = { ...items[itemIndex], ...overrides};
+          }
+        }
+
+        // Verify all existing options are valid, and if not remove them
+        for(let i = items.length - 1; i >= 0; --i) {
+          const item = items[i];
+          if(item.isValid && field && !item.isValid(args.propertyRecord, field)) {
```

Comment:
Grigas informed me that there are cases where a field could be undefined without implying an error state, he gave the example of if the property is a struct or array. I think it makes sense then that `isValid`'s definition allows for the `field` argument to be conditional